### DISCUSSION TITLE
Powershell, retaining filename

### DISCUSSION THREAD

**Discussion by a84r7a3rga76fg** [ImageMagick/ImageMagick#5945] [Help]
### ImageMagick version

7.1.0-57

### Operating system, version and so on

Windows 10, Powershell Core

### Description

Why doesn't this work? I want the filename to be the same. `magick *.png -resize 40% *.jpg`

**Comment by snibgo** [comment]
That command tries to read all the PNG files into memory. Then it resizes all those memory images. Then it tries to write them all to a file named "\*.jpg". But a jpg file can record only one image, so it tries to name then \*-0.jpg, \*-1.jpg, etc. But filenames on your OS probably can't start with "\*", so the command fails.

Instead, I suggest using "magick mogrify", like this:
```
magick mogrify -resize 40% -format jpg *.png
```
See https://imagemagick.org/script/mogrify.php

**Comment by a84r7a3rga76fg** [comment]
Is it not possible to load one image at a time? I don't have much RAM

**Comment by snibgo** [comment]
Yes, that's what `magick mogrify` does.
